src/ut/test_handlers.cpp

```diff
@@ -146,3 +359,86 @@ TEST_F(HandlerTest, NoNodeFunctionalityTest)
   ASSERT_EQ(NULL, msg);
   delete msg; msg = NULL;
 };
+
+TEST_F(HandlerTest, NotPost)
```

Comment:
These new tests look really good - but can you add a sentence at the start of the tests explicitly saying what each one is testing?